osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -4,98 +4,281 @@
-            return individualStrain + overallStrain - CurrentStrain;
+            /* Strain Bias for End On Body States
+             * Releasing multiple notes is as easy as releasing one.
+             *
+             *             End on Body Bias
+             *                 ▲
+             *             1.0 ┼ - - - - -  ────────────
+             *                 │           /
+             *             0.5 ┼ - - - -  ▲   Sigmoid Curve
+             *                 │         /│
+             *             0.0─┼──────────┴───────────────► Closest End Time / ms
+             *                 │   release_threshold
+             *                 │
+             * State E3  [=============]
+             * State F3        [=======]
+             * This            [============================]
+             * closestEndTime  <------->
+             *
+             * The closestEndTime is the closest end time to the current LN's head in ms.
+             */
+            if (isEndOnBody)
+                endOnBodyBias = 1 / (1 + Math.Exp(0.5 * (release_threshold - closestEndTime)));
+
+            // Strain Weight for End After Tail States
+            if (isEndAfterTail)
+                endAfterTailWeight = 1.25;
+
+            /* Update Column & Global Strain
+             * 1) We decay the strain given deltaTime
+             * 2) Increase strain given note and surroundings
+             *
+             * (!) endOnBodyBias only in Global Strain is a design choice.
+             *
+             *                  Column  Global
+             *                  Strain  Strain
+             * Default        : +2      +1
+             * If EndAfterTail: +2.5    +1.25
+             * If EndOnBody   :         +1    + (0, 1)
+             * If Both        :         +1.25 + (0, 1.25)
+             */
+            columnStrains[column] = applyDecay(columnStrains[column], startTime - prevStartTimes[column], decay_base);
+            columnStrains[column] += 2 * endAfterTailWeight;
+            globalStrain = applyDecay(globalStrain, current.DeltaTime, global_decay_base);
+            globalStrain += (1 + endOnBodyBias) * endAfterTailWeight;
+
+            /* For notes at the same time (in a chord), strain is the maximum column strain + global strain out of those columns
+             *
+             * 1) Why we want the maxima is to make the largest strain in the chord column invariant
+             * 2) The other strains in the chord is not important due to Strain binning (see function remarks) taking the maximal strain.
+             *
+             * i.e. We want max(GS) + max(CS): Global Strain (GS), Column Strain (CS)
+             *
+             * Given a state with a 4 note chord with:
+             * - GS: [A, B, C, D]  <- Strains are different in chords, GS are added every note. A < B < C < D
+             * - CS: [1, 2, 3, 4]
```

Comment:
What are the letters and numbers here? Do they refer to the giant table at the start? Or are they something completely different? I am thoroughly lost at this point.

Reply:
I'll see how to improve the explanation, as mentioned in another reply, I'll use a README.md instead.